arch/cpu/arm/cortex-m/trustzone/tz-api.h

```diff
@@ -0,0 +1,166 @@
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE
+ * COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/*
+ * \file
+ *   TrustZone API for communication between zones.
+ * \author
+ *   Niclas Finne <niclas.finne@ri.se>
+ *   Nicolas Tsiftes <nicolas.tsiftes@ri.se>
+ */
+
+/** \addtogroup arm
+ * @{
+ */
+
+/**
+ * \defgroup trustzone TrustZone for Arm Cortex-M
+ *
+ * This subsystem implements TrustZone support for Arm Cortex-M
+ * processors. The archtiecture is based on dual Contiki-NG firmwares:
+ * the secure world contains an instance of Contiki-NG with reduced
+ * functionality, and the normal world contains an instance with
+ * regular functionality. When programming an IoT device, the hex
+ * files with the two firmwares are merged into a single hex file,
+ * which is flashed to the device.
+ *
+ * Both worlds can access core system functionality such as processes,
+ * timers, and library functions. The normal world is expected to
+ * contain applications and networking functionality. By contrast, the
+ * secure world will contain secret information and functionality for
+ * monitoring the normal world. Hardware peripherals can be configured
+ * to be accessible in either of the worlds.
+ *
+ * Currently, the only supported Contiki-NG platform is the nRF5340
+ * development kit, which is equipped with two different Arm
+ * Cortex-M33 processors.
+ *
+ * @{
+ */
+
+#ifndef TZ_API_H
+#define TZ_API_H
+
+#include <stdbool.h>
+#include <stdlib.h>
+
+#ifdef NRF_TRUSTZONE_SECURE
```

Comment:
Doesn't this need to be wrapped around each macro for Doxygen output to look reasonable?